addons/mrp/migrations/12.0.2.0/post-migration.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2019 Eficent <http://www.eficent.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+from psycopg2.extensions import AsIs
+from odoo.addons.mrp import _create_warehouse_data
+
+
+def fill_mrp_workcenter_resource_calendar_id(cr):
+    openupgrade.logged_query(
+        cr, """
+        UPDATE mrp_workcenter mw
+        SET resource_calendar_id = rr.calendar_id
+        FROM resource_resource rr
+        WHERE mw.resource_id = rr.id AND mw.resource_calendar_id IS NULL
+        """
+    )
+
+
+def fill_mrp_workcenter_productivity_loss_loss_id(cr):
+    openupgrade.logged_query(
+        cr, """
+        UPDATE mrp_workcenter_productivity_loss wpl
+        SET loss_id = wplt.id
+        FROM mrp_workcenter_productivity_loss_type wplt
+        WHERE wpl.loss_id IS NULL AND wpl.%s = wplt.loss_type
+        """, (AsIs(openupgrade.get_legacy_name('loss_type')), ),
+    )
+
+
+def fill_stock_warehouse_picking_types(env):
+    warehouses = env['stock.warehouse'].search([])
+    for warehouse in warehouses:
+        warehouse.write(
+            warehouse._create_or_update_sequences_and_picking_types())
```

Comment:
Why are you doing this here? If you want to overwrite all picking types, then do it on `stock` end-migration.

Reply:
to have `pbm_type_id` field filled.

Reply:
Why in the end-migration of stock and not here?

Reply:
It breaks in `purchase_stock` if we do this in an end-migration. See https://github.com/OCA/OpenUpgrade/pull/1683 comment.

Reply:
I was asking that for avoiding calling several times to this method if there are other modules adding sequences like this. Seeing the problem, it's OK to do it here, but as always, please put an explanatory comment saying why (like the comment you have made).

Reply:
And also comment about `pbm_type_id`.